util/oidc/oidc.go

```diff
@@ -367,12 +387,49 @@ func (a *ClientApp) HandleCallback(w http.ResponseWriter, r *http.Request) {
 	if a.secureCookie {
 		flags = append(flags, "Secure")
 	}
-	var claims jwt.MapClaims
+	claims := Claims{}
 	err = idToken.Claims(&claims)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	groups := claims.Groups
+
+	// Some SSO implementations (Okta) require a call to
+	// the OIDC user info path to get attributes like groups
+	if a.userInfoPath != "" {
+		userInfo, err := a.GetUserInfo(token.AccessToken, a.settings.IssuerURL(), a.userInfoPath)
+		if err != nil {
+			w.WriteHeader(401)
+			_, _ = w.Write([]byte(fmt.Sprintf("failed to get groups claim: %v", err)))
+			return
+		}
+		if idToken.Subject != userInfo.Subject {
+			log.Warning("Subject of Claims from UserInfo endpoint didn't match Subject of idToken, drop usage of the UserInfo endpoint data")
+			return
+		}
```

Comment:
Could we go ahead and link the spec in this error message? https://openid.net/specs/openid-connect-core-1_0.html#UserInfo

It'll serve to document why the check exists and also to help someone who receives that error understand why they received it.

Reply:
Sure, I added the link to the error message. Definitely makes sense.